### PR TITLE
feat!: pass-through response status and headers to client

### LINKED ISSUE #25: Allow headers to pass-through to the response on the client
### Describe the feature

When using API-Party and making a request to a backend, default functionality should be for headers to pass-through to the client, and maybe have an ability filter them.

On the below images, we see one of the request responses using the native Fetch function sending a request directly to the endpoint, and the other using API Party.

I checked if any feature related to this issue existed in API Party but didn't find anything. 

image
image

If this is a non-feature, let me know how can I solve this in any other way.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Can you think of other implementations of this feature?

### Final checks

- [X] Check existing [issues](https://github.com/johannschopplich/nuxt-api-party/issues).

### PR DESCRIPTION
**Pull request by killjoy1221**
Breaking Change

### 🔗 Linked issue
Resolves #25 and #41

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
This change forwards the status code and headers from the API to the client. It also ensures any api errors are forwarded to the client as is.

Since ofetch always reads the body no matter what, it has to be read in as an `ArrayBuffer` then converted to an `Uint8Array`.

An additional side-effect of this is the decoded body is forwarded instead of raw, which means if a `content-encoding` response header is set, it will be wrong. Removing this header corrects that.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by johannschopplich** [comment]
I love this approach. It should suit most use-cases and is worth a breaking change. Well done. 👏

File: src/runtime/server/handler.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import { createError, defineEventHandler, getRequestHeader, getRouterParam, readBody } from 'h3'
```
Comment (johannschopplich) [inline src/runtime/server/handler.ts:1]: `setResponseStatus` and `setResponseHeaders` will be auto-imported locally, but break the distributed module. Please add the missing imports. (It's a bit confusing during development)

**Comment by johannschopplich** [comment]
Have you tested `$fetch.raw` behaves the same in other environments, like Cloudflare? Will test it a bit myself as well.

**Comment by killjoy1221** [comment]
I don't see why it would behave differently. If anything, putting it behind cloudflare would either suppress or expose the issue if it re-compresses the response. Honestly, the only reason I noticed this was because jsonplaceholder comments had the header `content-encoding: br`

**Pull request merged**